crates/bevy_pbr/src/meshlet/gpu_scene.rs

```diff
@@ -238,29 +244,28 @@ pub fn prepare_meshlet_per_frame_resources(
         &render_device,
         &render_queue,
     );
-    upload_storage_buffer(
-        &mut gpu_scene.previous_cluster_ids,
-        &render_device,
-        &render_queue,
-    );
+
+    // Early submission for GPU data uploads to start while the render graph records commands
+    render_queue.submit([]);
 
     let needed_buffer_size = 4 * gpu_scene.scene_triangle_count;
-    let visibility_buffer_draw_index_buffer =
-        match &mut gpu_scene.visibility_buffer_draw_index_buffer {
+    let visibility_buffer_draw_triangle_buffer =
+        match &mut gpu_scene.visibility_buffer_draw_triangle_buffer {
             Some(buffer) if buffer.size() >= needed_buffer_size => buffer.clone(),
             slot => {
                 let buffer = render_device.create_buffer(&BufferDescriptor {
-                    label: Some("meshlet_visibility_buffer_draw_index_buffer"),
+                    label: Some("meshlet_visibility_buffer_draw_triangle_buffer"),
                     size: needed_buffer_size,
-                    usage: BufferUsages::STORAGE | BufferUsages::INDEX,
+                    usage: BufferUsages::STORAGE,
                     mapped_at_creation: false,
                 });
                 *slot = Some(buffer.clone());
                 buffer
             }
         };
 
-    let needed_buffer_size = gpu_scene.scene_meshlet_count.div_ceil(32) as u64 * 4;
+    let needed_buffer_size =
+        gpu_scene.scene_meshlet_count.div_ceil(32) as u64 * size_of::<u32>() as u64;
```

Comment:
https://doc.rust-lang.org/std/primitive.u32.html#associatedconstant.BITS
```suggestion
        gpu_scene.scene_meshlet_count.div_ceil(u32::BITS) as u64 * size_of::<u32>() as u64;
```